testdata/testmodel/eval.go

```diff
@@ -20,10 +21,16 @@ type EvalTestSuite struct {
 }
 
 type EvalTest struct {
-	Name      string                    `json:"name"`
-	FlagKey   string                    `json:"flagKey"`
-	User      lduser.User               `json:"user"`
-	ValueType servicedef.ValueType      `json:"valueType"`
-	Default   ldvalue.Value             `json:"default"`
-	Expect    ldreason.EvaluationDetail `json:"expect"`
+	Name      string               `json:"name"`
+	FlagKey   string               `json:"flagKey"`
+	User      lduser.User          `json:"user"`
+	ValueType servicedef.ValueType `json:"valueType"`
+	Default   ldvalue.Value        `json:"default"`
+	Expect    ValueDetail          `json:"expect"`
```

Comment:
I changed the type here from `ldreason.EvaluationDetail` to a very similar new struct type `ValueDetail` so that 1. we don't have interoperability problems between `ldvalue.OptionalInt` (the type of `EvaluationDetail.VariationIndex) and `Maybe[int]`, and 2. on general principle, it's best to avoid relying on implicit behaviors of SDK types (e.g. their JSON property names).